packages/model-viewer/src/features/annotation.ts

```diff
@@ -0,0 +1,177 @@
+    for (let i = 0; i < 3; ++i) {
+      this.position.setComponent(
+          i, normalizeUnit(positionNodes[i] as NumberNode<'m'>).number);
+    }
+  }
+
+  updateNormal(normal?: string) {
+    if (normal == null)
+      return;
+    const normalNodes = parseExpressions(normal)[0].terms;
+    for (let i = 0; i < 3; ++i) {
+      this.normal.setComponent(
+          i, normalizeUnit(normalNodes[i] as NumberNode<'m'>).number);
+    }
+  }
+}
+
+export declare interface AnnotationInterface {
+  updateHotspot(config: HotspotConfiguration): void;
+}
+
+export const AnnotationMixin = <T extends Constructor<ModelViewerElementBase>>(
+    ModelViewerElement: T): Constructor<AnnotationInterface>&T => {
+  class AnnotationModelViewerElement extends ModelViewerElement {
+    private[$annotationRenderer] = new CSS2DRenderer();
+    private[$hotspotMap] = new Map();
+    private[$observer] = new MutationObserver((mutations) => {
+      mutations.forEach((mutation) => {
+        if (mutation.type === 'childList') {
+          mutation.addedNodes.forEach((node) => {
+            this[$addHotspot](node);
+          });
+          mutation.removedNodes.forEach((node) => {
+            this[$removeHotspot](node);
+          });
+        }
+      });
+    });
+
+    connectedCallback() {
+      super.connectedCallback();
+      const {domElement} = this[$annotationRenderer];
+      const {style} = domElement;
+      style.pointerEvents = 'none';
+      style.position = 'absolute';
+      style.top = '0';
+      this.shadowRoot!.querySelector('.container')!.appendChild(domElement);
+
+      for (let i = 0; i < this.children.length; ++i) {
+        this[$addHotspot](this.children[i]);
+      }
+
+      this[$observer].observe(this, {childList: true});
+    }
+
+    disconnectedCallback() {
+      super.disconnectedCallback();
+      this[$observer].disconnect();
+    }
+
+    updated(changedProperties: Map<string, any>) {
+      super.updated(changedProperties);
+    }
+
+    [$tick](time: number, delta: number) {
+      super[$tick](time, delta);
+      this[$updateHotspots]();
+      this[$annotationRenderer].render(this[$scene], this[$scene].activeCamera);
+    }
+
+    [$onResize](e: {width: number, height: number}) {
+      super[$onResize](e);
+      this[$annotationRenderer].setSize(e.width, e.height);
+    }
+
+    [$updateHotspots]() {
+      const {children} = this[$scene].pivot;
+      for (let i = 0, l = children.length; i < l; i++) {
+        const object = children[i];
+        if (object instanceof Hotspot) {
+          const view = this[$scene].activeCamera.position.clone();
+          view.sub(object.position);
+          if (view.dot(object.normal) < 0) {
+            object.element.classList.add('hide');
+          } else {
+            object.element.classList.remove('hide');
+          }
+        }
+      }
+    }
+
+    [$addHotspot](node: Node) {
+      if (!(node instanceof HTMLElement && node.slot.includes('hotspot')))
```

Comment:
For the purposes of IE11 support, we should avoid `includes`. Consider `indexOf('hotspot') === 0` as an alternative.